packages/datadog-instrumentations/src/oracledb.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const connectionAttributes = new WeakMap()
+const poolAttributes = new WeakMap()
+
+const executeStartChannel = channel('apm:oracledb:execute:start')
+const executeErrorChannel = channel('apm:oracledb:execute:error')
+const executeEndChannel = channel('apm:oracledb:execute:end')
+
+function finish (err) {
+  if (err) {
+    executeErrorChannel.publish(err)
+  }
+  executeEndChannel.publish(undefined)
+}
+
+addHook({ name: 'oracledb', versions: ['5'] }, oracledb => {
+  shimmer.wrap(oracledb.Connection.prototype, 'execute', execute => {
+    return function wrappedExecute (dbQuery, ...args) {
+      if (!executeStartChannel.hasSubscribers) {
+        return execute.apply(this, arguments)
+      }
+
+      if (arguments.length && typeof arguments[arguments.length - 1] === 'function') {
+        const cb = arguments[arguments.length - 1]
+        const outerAr = new AsyncResource('oracledb:outer-scope')
+        arguments[arguments.length - 1] = function wrappedCb (err, result) {
+          finish(err)
+          return outerAr.runInAsyncScope(() => cb.apply(this, arguments))
+        }
+      }
+
+      return AsyncResource.bind(() => {
+        const connAttrs = connectionAttributes.get(this)
+        executeStartChannel.publish({ query: dbQuery, connAttrs })
+        try {
+          const result = execute.apply(this, arguments)
+
+          if (result && typeof result.then === 'function') {
+            result.then(() => finish(), finish)
```

Comment:
This should be reassigned or returned to avoid swallowing errors.

Reply:
I'll do that, making sure to propagate the resolved value or error properly.